zerver/actions/message_edit.py

```diff
@@ -1411,24 +1411,28 @@ def check_update_message(
         isinstance(message_edit_request, StreamMessageEditRequest)
         and message_edit_request.is_topic_edited
     ):
-        if not user_profile.can_move_messages_to_another_topic():
-            raise JsonableError(_("You don't have permission to edit this message"))
+        if message_edit_request.topic_resolved or message_edit_request.topic_unresolved:
+            if not user_profile.can_resolve_topic():
+                raise JsonableError(_("You don't have permission to resolve topics."))
+        else:
```

Comment:
I think this `else` might be wrong given how `build_message_edit_request` will set `topic_resolved` if the topic is resolved and the channel is also changed (with no other change to the topic). This needs to be resolved and tested.

Reply:
Oh, no, that's handled separately, in the `is_stream_edited` block. OK.